File: 06_asymmetric/rsa/01_pkcs1_keys/02_oaep_padding/encrypt.cs
using System.Security.Cryptography;
using System.Text;

var rsa = RSA.Create();
rsa.ImportFromPem(File.ReadAllText("public.pem"));

var text = "Hello world, mac was here";

var encrypted = rsa.Encrypt(Encoding.UTF8.GetBytes(text), RSAEncryptionPadding.OaepSHA1);

Console.WriteLine(Convert.ToBase64String(encrypted));

File.WriteAllText("protected.txt", Convert.ToBase64String(encrypted));

// RSAEncryptionPadding.Pkcs1      - PKCS #1 encryption standard
// RSAEncryptionPadding.OaepSHA1   - Optimal Asymmetric Encryption Padding (OAEP) encryption standard with a SHA1 hash algorithm
// RSAEncryptionPadding.OaepSHA256 - Optimal Asymmetric Encryption Padding (OAEP) encryption standard with a SHA256 hash algorithm
// RSAEncryptionPadding.OaepSHA384 - Optimal Asymmetric Encryption Padding (OAEP) encryption standard with a SHA-384 hash algorithm
// RSAEncryptionPadding.OaepSHA512 - Optimal Asymmetric Encryption Padding (OAEP) encryption standard with a SHA512 hash algorithm

File: 06_asymmetric/rsa/01_pkcs1_keys/02_oaep_padding/encrypt.php
<?php
$text = "Hello world, mac was here";

openssl_public_encrypt($text, $encrypted, file_get_contents("public.pem"), OPENSSL_PKCS1_OAEP_PADDING);	

$encoded = base64_encode($encrypted);

echo $encoded . PHP_EOL;

file_put_contents("protected.txt", $encoded);

// OPENSSL_PKCS1_PADDING, OPENSSL_SSLV23_PADDING, OPENSSL_PKCS1_OAEP_PADDING, OPENSSL_NO_PADDING.